src/path_rewriting.rs

```diff
@@ -97,6 +99,30 @@ fn get_abs_path(source_dir: &Option<PathBuf>, rel_path: PathBuf) -> (PathBuf, Pa
     (abs_path, rel_path)
 }
 
+fn reverse(mut s: String) -> String {
+    unsafe {
+        let vec = s.as_mut_vec();
```

Comment:
if the string contains utf-8 chars, then this code is bad.
You should use something like:
```rust
fn reverse(mut s: String) -> String {
    s.chars().rev().collect::<String>()
}
```